src/dmd/root/filename.d

```diff
@@ -894,4 +895,82 @@ version(Windows)
 
         return F(ret);
     }
+
+}
+
+version(Posix)
+{
+
+    /**************************
+    Takes a callable F and applies it to the result of converting
+    `fileName` to an absolute file path (char*)
+    Params:
+        fileName = The file name to be converted to an absolute path
+    Returns: Whatever `F` returns.
+    */
+    auto absPathThen(alias F)(const(char)* fileName)
+    {
+        import core.sys.posix.stdlib: realpath;
+        char[1024] realPathBuffer;
```

Comment:
The maximal path length on Posix is typically at least 4096!

Reply:
It turns out that realpath can allocate so I changed it to do that.